### PR TITLE
Increase coverage

### PR DESCRIPTION
**Pull request by Goodmain**

### CODE REVIEW

File: src/Services/HttpRequestService.php [hunk 1]
```diff
@@ -157,6 +157,9 @@ protected function sendRequest($method, $url, array $data = [], array $headers =
         return $response;
     }
 
+    /**
+     * @codeCoverageIgnore
+     */
```
Comment (DenTray) [inline src/Services/HttpRequestService.php:162]: could we mock logger instead?

File: src/Traits/MailsMockTrait.php [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+trait MailsMockTrait
+{
+    use FixturesTrait;
+
+    private array $requiredExpectationParameters = [
+        'emails',
+        'fixture'
+    ];
+    /**
```
Comment (DenTray) [inline src/Traits/MailsMockTrait.php:19]:
```suggestion

    /**
```

File: src/Traits/MailsMockTrait.php [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    }
+
+    protected function validateExpectationParameters(array $currentMail, int $index): void
+    {
+        foreach ($this->requiredExpectationParameters as $parameter) {
+            if (!Arr::has($currentMail, $parameter)) {
+                abort(
+                    Response::HTTP_INTERNAL_SERVER_ERROR,
+                    "Missing required key \"{$parameter}\" in the input data set on the step: {$index}."
+                );
```
Comment (DenTray) [inline src/Traits/MailsMockTrait.php:91]:
```suggestion
                $this->fail("Missing required key \"{$parameter}\" in the input data set on the step: {$index}.");
```

File: src/Traits/MailsMockTrait.php [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+            $index,
+            "Failed assert that send emails count are equals, expected send email count: {$countData}, actual {$index}."
+        );
+    }
+
+    protected function assertEmailsList(array $expectedMailData, Mailable $mail, int $index): void
+    {
+        $sentEmails = Arr::pluck($mail->to, 'address');
+        $emails = Arr::wrap($expectedMailData['emails']);
+        $this->assertAddressesCount($emails, $mail, $index);
```
Comment (DenTray) [inline src/Traits/MailsMockTrait.php:163]:
```suggestion

        $this->assertAddressesCount($emails, $mail, $index);
```

File: src/Traits/MailsMockTrait.php [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    protected function prepareEmailChain($emailChain): array
+    {
+        if (is_string($emailChain)) {
+            $emailChain = $this->getJsonFixture($emailChain);
+        }
+
+        return (is_multidimensional($emailChain)) ? $emailChain : [$emailChain];
+    }
+
+    protected function sentMail($emails, string $fixture, string $subject = '', $from = ''): array
```
Comment (DenTray) [inline src/Traits/MailsMockTrait.php:193]:
```suggestion
    protected function mockedMail($emails, string $fixture, string $subject = '', $from = ''): array
```

File: src/Traits/ModelTrait.php [hunk 6]
```diff
@@ -43,7 +43,8 @@ protected function getRelationshipFromMethod($method)
             $modelName = static::class;
 
             throw new BadMethodCallException(
-                "Attempting to lazy-load relation '{$method}' on model '{$modelName}'. See property \$disableLazyLoading"
+                "Attempting to lazy-load relation '{$method}' "
+                . "on model '{$modelName}'. See property \$disableLazyLoading"
```
Comment (DenTray) [inline src/Traits/ModelTrait.php:47]:
```suggestion
                "Attempting to lazy-load relation '{$method}' on model '{$modelName}'."
                . ' See property \$disableLazyLoading"';
```

File: tests/HttpRequestServiceTest.php [hunk 7]
```diff
@@ -305,4 +310,22 @@ public function testNotJSONResponse()
 
         $this->httpRequestServiceClass->json();
     }
+
+    public function testSendWithRequestException()
+    {
+        $this->expectException(RequestException::class);
```
Comment (DenTray) [inline tests/HttpRequestServiceTest.php:316]: could we also expect exception message by using `expectExceptionMessage`

File: tests/fixtures/MailsMockTraitTest/.gitignore [hunk 8]
```diff
@@ -0,0 +1 @@
+test_mail_with_export.html
```
Comment (DenTray) [inline tests/fixtures/MailsMockTraitTest/.gitignore:1]: ?

**Review by DenTray — CHANGES_REQUESTED** [review]

File: src/Traits/MailsMockTrait.php (hunk 2 above)
Comment (Goodmain) [reply src/Traits/MailsMockTrait.php:19]: Done

File: src/Services/HttpRequestService.php (hunk 1 above)
Comment (Goodmain) [reply src/Services/HttpRequestService.php:162]: I added deprecation and text that better to use additional packages for logging

File: src/Traits/MailsMockTrait.php (hunk 4 above)
Comment (Goodmain) [reply src/Traits/MailsMockTrait.php:163]: Done

File: src/Traits/MailsMockTrait.php (hunk 5 above)
Comment (Goodmain) [reply src/Traits/MailsMockTrait.php:193]: Done

File: tests/fixtures/MailsMockTraitTest/.gitignore (hunk 8 above)
Comment (Goodmain) [reply tests/fixtures/MailsMockTraitTest/.gitignore:1]: It is a temporary file which we creating in `testMailWithExport`.

File: tests/HttpRequestServiceTest.php (hunk 7 above)
Comment (Goodmain) [reply tests/HttpRequestServiceTest.php:316]: Done

File: src/Traits/ModelTrait.php (hunk 6 above)
Comment (Goodmain) [reply src/Traits/ModelTrait.php]: Done

File: src/Traits/MailsMockTrait.php (hunk 3 above)
Comment (Goodmain) [reply src/Traits/MailsMockTrait.php]: Done

**Review by DenTray — CHANGES_REQUESTED** [review]

**Pull request merged**